crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -479,962 +673,117 @@ impl CheckpointExecutor {
+            .await
+            .expect("Failed to accumulate running root");
+
+        if ckpt_state.data.checkpoint.is_last_checkpoint_of_epoch() {
+            self.checkpoint_store
+                .insert_epoch_last_checkpoint(self.epoch_store.epoch(), &ckpt_state.data.checkpoint)
+                .expect("Failed to insert epoch last checkpoint");
+
             self.accumulator
-                .accumulate_running_root(epoch_store, checkpoint.sequence_number, checkpoint_acc)
-                .await
-                .expect("Failed to accumulate running root");
-            self.bump_highest_executed_checkpoint(checkpoint);
+                .accumulate_epoch(self.epoch_store.clone(), seq)
+                .expect("Accumulating epoch cannot fail");
+
+            self.checkpoint_store
+                .prune_local_summaries()
+                .tap_err(|e| debug_fatal!("Failed to prune local summaries: {}", e))
+                .ok();
         }
+
+        fail_point!("crash");
+
+        self.bump_highest_executed_checkpoint(&ckpt_state.data.checkpoint);
     }
 
     /// If configured, commit the pending index updates for the provided checkpoint as well as
     /// enqueuing the checkpoint to the subscription service
-    async fn commit_index_updates_and_enqueue_to_subscription_service(
-        &self,
-        checkpoint: CheckpointData,
-    ) {
+    fn commit_index_updates_and_enqueue_to_subscription_service(&self, checkpoint: CheckpointData) {
         if let Some(rpc_index) = &self.state.rpc_index {
             rpc_index
                 .commit_update_for_checkpoint(checkpoint.checkpoint_summary.sequence_number)
                 .expect("failed to update rpc_indexes");
         }
 
         if let Some(sender) = &self.subscription_service_checkpoint_sender {
-            if let Err(e) = sender.send(checkpoint).await {
-                tracing::warn!("unable to send checkpoint to subscription service: {e}");
+            if let Err(e) = sender.blocking_send(checkpoint) {
+                warn!("unable to send checkpoint to subscription service: {e}");
             }
         }
     }
 
-    #[instrument(level = "debug", skip_all)]
-    fn schedule_synced_checkpoints(
-        &self,
-        pending: &mut CheckpointExecutionBuffer,
-        next_to_schedule: &mut CheckpointSequenceNumber,
-        epoch_store: Arc<AuthorityPerEpochStore>,
-        run_with_range: Option<RunWithRange>,
-    ) {
-        let Some(latest_synced_checkpoint) = self
-            .checkpoint_store
-            .get_highest_synced_checkpoint()
-            .expect("Failed to read highest synced checkpoint")
-        else {
-            debug!("No checkpoints to schedule, highest synced checkpoint is None",);
-            return;
-        };
+    // Write checkpoint data to indexes and disk (if configured)
+    fn write_checkpoint_data(&self, checkpoint_data: &CheckpointData) {
```

Comment:
Do we want to rename this to be more accurate to what it is doing since it can either write to disk or index or both if configured as such.

Reply:
i just inlined this function at the callsite since the calling function was already quite small